src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -807,5 +816,48 @@ internal sealed class LoggerParameter
             // but instead is supposed to be taken as a parameter for the template.
             public bool IsTemplateParameter => !IsLogger && !IsException && !IsLogLevel;
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static uint RotateLeft(uint value, int offset) => (value << offset) | (value >> (32 - offset));
+
+        /// <summary>
+        /// Returns a non-randomized hash code for the given string.
+        /// This implementation is porting the one used in the core
+        /// https://github.com/dotnet/runtime/blob/f3b599e5777eb1db7c396fdc1597c390f22879e6/src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs#L815
+        /// We always return a positive value.
+        /// </summary>
+        private static int GetNonRandomizedHashCode(string source)
```

Comment:
This is ok, but it'd also be ok to just use a really simple one like fnv that Roslyn uses for switches, e.g.
[SharpLab](https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoBBmABM3AYVwG9NcL89gB7GgG1wFkAKBFABlwGcBKcymQyUR+AOw9cUbgNGiARBCUR5uGgCdcipcFUatOlWs2LgSvSbPnjsuZW3Bdxg46P75rtyYhmn73yoA3LYAvpghQA===)
```C#
internal static uint ComputeStringHash(string s)
{
    uint result = 2166136261u;
    foreach (char c in s)
    {
        result = (c ^ result) * 16777619;
    }
    return result;
}
```

Reply:
Ok, I'll use the simple one.